install/kubernetes/helm/istio/values.yaml

```diff
@@ -316,6 +316,8 @@ security:
   image: citadel
   selfSigned: true # indicate if self-signed CA is used.
   cleanUpOldCA: true
+  # Enabled Health Check will add the livenessProbe to the Citadel container
+  healthCheckEnabled: false
```

Comment:
@rshriram Just to match how it was before. There is a [security task](https://preliminary.istio.io/docs/tasks/security/health-check.html) to enable this optional health check but I see your point of just having it part of default without flag.

@wattli @lei-tang Do you see any issue with having the livenessProbe part of default with no need to go through a task to learn how to enable it? It will make the task a verification/customization task.

Reply:
I do not see an issue of having health check on by default. But the section "(Optional) Configuring the health checking" (https://preliminary.istio.io/docs/tasks/security/health-check.html#optional-configuring-the-health-checking) still uses istio-citadel-with-health-check.yaml. I am not sure that adding the healthCheckEnabled flag will address the missing file problem for the section "(Optional) Configuring the health checking".

Reply:
@lei-tang The idea is to remove the flag and always have the health check (livenessProve) in the container so users don't need to go through a task to configure/add it.

Reply:
Removing/adding the healthCheckEnabled flag does not solve the problem reported in #6922 because the the section "(Optional) Configuring the health checking" (https://preliminary.istio.io/docs/tasks/security/health-check.html#optional-configuring-the-health-checking) still uses istio-citadel-with-health-check.yaml to configure the parameters such as liveness-probe-interval, probe-check-interval, and etc. These parameters are orthogonal to the healthCheckEnabled flag.

Reply:
It's not orthogonal because Helm template generates an istio.yaml that have those paramters about `livenessProbe` and the users can modify those to fit their needs. Just like any customisation with all other components. Yes, the doc needs to be updated.

P.S. does it really deserve a task or is it just an installation option?

Reply:
@yangminzhu Do you see an issue?

Reply:
@liminw @wattli https://github.com/istio/istio/pull/6961#discussion_r201568792